src/pages/ProductListing/ProductListing.jsx

```diff
@@ -1,8 +1,43 @@
-import React from 'react'
+import React, { useEffect } from 'react'
+import axios from 'axios'
+import './ProductListing.css'
+import { ProductCard } from '../../components/ProductCard/ProductCard'
+import { useProduct } from '../../context/product-context'
+import { sortByPriceFunc , sortByRatingFunc , sortByCategoryFunc , priceRangeFilter } from '../../utilities/index'
+import { Nav } from '../../components/Nav/Nav'
+import { Filter } from '../../components/Filter/Filter'
 
 function ProductListing() {
+    const {state:{products, sortByPrice,sortByRating,sortByCategory,priceRange},dispatch} = useProduct()
+
+    useEffect(() => {
+        (async function fetchProducts() {
+            try {
+                const {data} = await axios.get("/api/products")
+                dispatch({type: "LOAD-PRODUCTS", payload: data.products})
+            } catch(error) {
+                alert("error from server" , error)
+            } 
+        })()
+    },[])
+
+    let sortedProducts = products
+    sortedProducts = sortByPriceFunc(products,sortByPrice)
+    let sortByRatingProducts = sortByRatingFunc(sortedProducts,sortByRating)
+    let sortByCategoryProducts = sortByCategoryFunc(sortByRatingProducts,sortByCategory)
+    let filteredPriceProducts = priceRangeFilter(sortByCategoryProducts,priceRange)
```

Comment:
Try to use `const` at every place.

Reply:
tried to use but it was not working. will look into it after completion